src/Interpretation.agda

```diff
@@ -55,39 +58,38 @@ record Model ℓ m : Set (suc ℓ ⊔ suc m) where
     ⟦extend⟧ : ∀ {X Y Z} → ((X ⟦×⟧ Y) ==> Mon Z) → (X ⟦×⟧ Mon Y) ==> Mon Z
 
     -- Actual stuff
-    ⟦Bool⟧ : LinearityVal → PolarityVal → ⟦Type⟧
-    ⟦Num⟧  : LinearityVal → ⟦Type⟧
+    ⟦Bool⟧ : BoolRestriction → ⟦Type⟧
+    ⟦Num⟧  : NumRestriction → ⟦Type⟧
 
-    ⟦not⟧ : ∀ {l p₁ p₂} → (Flat (NegPolRel p₁ p₂) ⟦×⟧ ⟦Bool⟧ l p₁) ==> ⟦Bool⟧ l p₂
-    ⟦and⟧ ⟦or⟧ : ∀ {l₁ l₂ l₃ p₁ p₂ p₃} →
-                (Flat (MaxLinRel l₁ l₂ l₃) ⟦×⟧
-                 (Flat (MaxPolRel p₁ p₂ p₃) ⟦×⟧
-                  (⟦Bool⟧ l₁ p₁ ⟦×⟧ ⟦Bool⟧ l₂ p₂))) ==> ⟦Bool⟧ l₃ p₃
+    ⟦not⟧ : ∀ {b₁ b₂} → (Flat (NotRestriction b₁ b₂) ⟦×⟧ ⟦Bool⟧ b₁) ==> ⟦Bool⟧ b₂
+    ⟦and⟧ : ∀ {b₁ b₂ b₃} → (Flat (AndRestriction b₁ b₂ b₃) ⟦×⟧ (⟦Bool⟧ b₁ ⟦×⟧ ⟦Bool⟧ b₂)) ==> ⟦Bool⟧ b₃
+    ⟦or⟧ : ∀ {b₁ b₂ b₃} → (Flat (OrRestriction b₁ b₂ b₃) ⟦×⟧ (⟦Bool⟧ b₁ ⟦×⟧ ⟦Bool⟧ b₂)) ==> ⟦Bool⟧ b₃
 
-    ⟦if⟧ : ∀ {X} → ((Mon X ⟦×⟧ Mon X) ⟦×⟧ ⟦Bool⟧ linear U) ==> Mon X
+    ⟦if⟧ : ∀ {X b} → ((Mon X ⟦×⟧ Mon X) ⟦×⟧ (Flat (IfRestriction b) ⟦×⟧ (⟦Bool⟧ b))) ==> Mon X
 
-    ⟦∃⟧    : ∀ {p₁ p₂ l} →
-            (Flat (QuantifyRel p₁ p₂) ⟦×⟧ (⟦Num⟧ linear ⟦⇒⟧ Mon (⟦Bool⟧ l p₁))) ==> ⟦Bool⟧ l p₂
+    ⟦∃⟧    : ∀ {l b₁ b₂} → (Flat (QuantRestriction l b₁ b₂) ⟦×⟧ (⟦Num⟧ l ⟦⇒⟧ Mon (⟦Bool⟧ b₁))) ==> ⟦Bool⟧ b₂
 
-    ⟦add⟧     : ∀ {l₁ l₂ l₃} → (Flat (MaxLinRel l₁ l₂ l₃) ⟦×⟧ (⟦Num⟧ l₁ ⟦×⟧ ⟦Num⟧ l₂)) ==> ⟦Num⟧ l₃
-    ⟦mul⟧     : ∀ {l₁ l₂ l₃} → (Flat (MulRel l₁ l₂ l₃) ⟦×⟧ (⟦Num⟧ l₁ ⟦×⟧ ⟦Num⟧ l₂)) ==> ⟦Num⟧ l₃
-    ⟦const⟧   : ∀ {X} → ℚ → X ==> ⟦Num⟧ const
-    ⟦extFunc⟧ : ⟦Num⟧ linear ==> Mon (⟦Num⟧ linear)
-    ⟦≤⟧       : ∀ {l₁ l₂ l₃} → (Flat (MaxLinRel l₁ l₂ l₃) ⟦×⟧ (⟦Num⟧ l₁ ⟦×⟧ ⟦Num⟧ l₂)) ==> ⟦Bool⟧ l₃ U
+    ⟦add⟧     : ∀ {l₁ l₂ l₃} → (Flat (AddRestriction l₁ l₂ l₃) ⟦×⟧ (⟦Num⟧ l₁ ⟦×⟧ ⟦Num⟧ l₂)) ==> ⟦Num⟧ l₃
+    ⟦mul⟧     : ∀ {l₁ l₂ l₃} → (Flat (MulRestriction l₁ l₂ l₃) ⟦×⟧ (⟦Num⟧ l₁ ⟦×⟧ ⟦Num⟧ l₂)) ==> ⟦Num⟧ l₃
+    ⟦const⟧   : ∀ {l₁} → ℚ → (Flat (NumConstRestriction l₁)) ==> ⟦Num⟧ l₁
```

Comment:
This is the one change I wasn't sure about. The old type was `∀ {X} → ℚ → X ==> ⟦Num⟧ const`, but I really couldn't figure out the purpose of the arbitrary type `X`, nor how to adapt the proof so it's still included. All the proofs seem to work without it.

Is this change okay, or have I broken something we're modelling?